wsmaster/che-core-api-installer/src/main/java/org/eclipse/che/api/installer/server/spi/InstallerDao.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * <p><strong>Note:</strong> This particularly does not mean that
+ * method call will not make any inconsistency, but this mean that
+ * such kind of inconsistencies are expected by design and may be treated further.
+ *
+ * @author Anatolii Bazko
+ */
+public interface InstallerDao {
+
+    /**
+     * Creates a new installer.
+     *
+     * @param installer
+     *         installer to create
+     * @throws NullPointerException
+     *         when {@code installer} is null
+     * @throws InstallerConflictException
+     *         when installer with such id and version already exists
+     * @throws InstallerException
+     *         when any other error occurs
+     */
+    void create(InstallerImpl installer) throws InstallerException;
+
+    /**
+     * Updates installer by replacing an existing entity with a new one.
+     *
+     * @param installer
+     *         installer to update
+     * @throws NullPointerException
+     *         when {@code installer} is null
+     * @throws InstallerNotFoundException
+     *         when installer with id and version doesn't exist
+     * @throws InstallerException
+     *         when any other error occurs
+     */
+    void update(InstallerImpl installer) throws InstallerException;
+
+    /**
+     * Removes installer.
+     *
+     * <p>Note that this method doesn't throw any exception when
+     * installer doesn't exist.
+     *
+     * @param fqn
+     *         unique installer identifier
+     * @throws NullPointerException
+     *         when {@code fqn} is null
+     * @throws InstallerException
+     *         when any other error occurs
+     */
+    void remove(InstallerFqn fqn) throws InstallerException;
+
+    /**
+     * Finds installer by its fqn.
+     *
+     * @param fqn
+     *         unique installer identifier
+     * @return installer instance, never null
+     * @throws NullPointerException
+     *         when {@code fqn} is null
+     * @throws InstallerNotFoundException
+     *         when installer with given {@code fqn} doesn't exist
+     * @throws InstallerException
+     *         when any other error occurs
+     */
+    InstallerImpl getByFqn(InstallerFqn fqn) throws InstallerException;
+
+    /**
+     * Gets all installers from persistent layer of the specific id.
+     * The version of the installer are not taken into account.
+     *
+     * @param id
+     *        the installer id
+     *        {@link InstallerFqn#id}
+     * @param maxItems
+     *         the maximum number of installers to return
+     * @param skipCount
+     *         the number of installers to skip
+     * @return list of installers or empty list if no installers were found
+     *         when {@code maxItems} or {@code skipCount} is negative
+     * @throws InstallerException
+     *         when any other error occurs
+     */
+    Page<InstallerImpl> getVersions(String id, int maxItems, long skipCount) throws InstallerException;
```

Comment:
`getById` name is supposed to return the only installer as opposite to `getVersions` name.